File: maximum-number-of-vowels/maximum-number-of-vowels.js
//Leetcode - Maximum number of vowels in a substring of given length

var maxVowels = function(s, k) {
    const vowels = new Set(['a', 'e', 'i', 'o', 'u']);
    let starter_count = 0;
    for (let i = 0; i < k; i++) {
        if (vowels.has(s[i])) {
            starter_count += 1
        }
    };
    let maximum_vowel = starter_count;
    let start = 1;
    let end = k;
    while (end < s.length) {
        if (vowels.has(s[end])) {
            starter_count += 1
        };
        if (vowels.has(s[start - 1])) {
            starter_count -= 1
        };
        if (starter_count > maximum_vowel) {
            maximum_vowel = starter_count
        };
        start += 1;
        end += 1
    };
    return maximum_vowel
};

File: maximum-number-of-vowels/maximum-number-of-vowels.py
#Leetcode - Maximum number of vowels in a substring of given length

def maxVowels(self, s: str, k: int) -> int:
        vowels = set(['a', 'e', 'i', 'o', 'u'])
        starter_count = 0
        for i in range(k):
            if s[i] in vowels:
                starter_count += 1
        max_count = starter_count
        end = k
        while end < len(s):
            if s[end] in vowels:
                starter_count += 1
            if s[end - k] in vowels:
                starter_count -= 1
            max_count = max(max_count, starter_count)
            end += 1
        return max_count